src/common/path_util.cpp

```diff
@@ -82,11 +82,19 @@ static std::filesystem::path GetBundleParentDirectory() {
 
 static auto UserPaths = [] {
 #ifdef __APPLE__
-    std::filesystem::current_path(GetBundleParentDirectory());
+    const auto base_dir = GetBundleParentDirectory();
+    const auto user_dir = base_dir / "user";
+    if (std::filesystem::exists(base_dir)) {
+        return std::filesystem::path{user_dir};
+    } else {
+        const auto base_dir = std::filesystem::path(getenv("HOME")) / "Library/Application Support/shadPS4";
+        const auto user_dir = base_dir / "user";
+            std::filesystem::create_directories(user_dir);
+            return user_dir;
```

Comment:
You can't return early here or in the other path, you're skipping all the extra logic for creating the subdirectories below.